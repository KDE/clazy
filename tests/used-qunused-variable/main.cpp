#include <QtCore/QObject>
#include <QtCore/QString>




void testUsedVar(QString bla)
{
    Q_UNUSED(bla) // WARN
    Q_UNUSED(0) // OK, just some other variable
    bla.at(0);
}

void testUnusedVar(QString bla)
{
    Q_UNUSED(bla) // OK, it is really unused
    int i = 1;
}

void testVoidStmt(const QString &bla)
{
  (void) bla; // WARN
  bla.at(0);
}
void testCastStmt(int bla)
{
  auto casted = (float) bla; // OK - diferent from void cast
}
