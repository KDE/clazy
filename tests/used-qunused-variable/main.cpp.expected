used-qunused-variable/main.cpp:9:14: warning: Q_UNUSED used even though variable has usages [-Wclazy-used-qunused-variable]
used-qunused-variable/main.cpp:22:10: warning: void cast used even though variable has usages [-Wclazy-used-qunused-variable]
